libraries/AP_IRLock/IRLock.h

```diff
@@ -60,13 +61,14 @@ class IRLock
     uint32_t _last_update_ms;
 
     // irlock_target_info is a duplicate of the PX4Firmware irlock_s structure
-    typedef struct {
+    struct irlock_target_info {
```

Comment:
Add the typedef back in. This is declaring a variable irlock_target_info with type struct irlock_target_info { ... }. What you want to do is define the type irlock_target_info as struct { ... }.

So it should be:
```
typedef struct {
    // ...
} irlock_target_info;
irlock_target_info _target_info[10];
```

Reply:
Done